lib/main.dart

```diff
@@ -17,41 +17,16 @@ import 'package:flutter/material.dart';
 
 import 'src/app.dart' deferred as app;
 
-void main() {
-  final Future<void> loadedLibrary = app.loadLibrary();
+void main() async {
+  // Initialize Firebase
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp();
+
+  final Future<void> loadedLibrary = await app.loadLibrary();
```

Comment:
These two can be merged without having to await for either:

```dart
final Future<void> loaded = Future.wait([app.loadLibrary(), Firebase.initializeApp()]);
```

Reply:
Looks better, I'll update it

Reply:
I just realized that the Future.wait resolves to the loadedLibrary as the 0th position of the array of results, so this can't be Future<void> (but you also don't need to specify the type on the left either).

Reply:
Oh I see. I was trying to do it and when the app launches there shows an [error screen](https://i.imgur.com/sNcQ3oZ.png) for like 1 sec and then it loads the content of the app

``` dart
final loaded = Future.wait([
    app.loadLibrary(),
    Firebase.initializeApp(),
  ]);
```

``` dart
The relevant error-causing widget was: 
  FutureBuilder<List<dynamic>> file:///C:/Users/matej/FlutterProjects/forked/codelab-friendlyeats-flutter/lib/main.dart:30:5
When the exception was thrown, this was the stack: 
#0      _checkLoaded (dart:core-patch/lib_prefix.dart:74:5)
#1      main.<anonymous closure> (package:friendlyeats/main.dart)
#2      _FutureBuilderState.build (package:flutter/src/widgets/async.dart:740:55)
#3      StatefulElement.build (package:flutter/src/widgets/framework.dart:4681:28)
#4      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4564:15)
```

Reply:
Does the `snapshot` of the FutureBuilder contain a List<dynamic>? If so, snapshot.data[0] should be the loaded library.

(Wait resolves to an array of resolved futures, in the same order you passed them to .wait)

Reply:
(I think the builder needs to check `if (snapshot.hasData)` before calling `app.FriendlyEatsApp()`)